### PR TITLE
NuGetAuthenticateV1: Add WIF Service Connection Support

### PR DESCRIPTION
**Pull request by embetten**
**Task name**:  NuGetAuthenticateV1

**Description**: Updating task to support Wif service connections

**Documentation changes required:** (Y/N) Y

**Added unit tests:** (Y/N) N/A

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/NuGetAuthenticateV1/main.ts [hunk 1]
```diff
@@ -1,29 +1,53 @@
...
         // Install the credential provider
         forceReinstallCredentialProvider = tl.getBoolInput("forceReinstallCredentialProvider", false);
         await installCredProviderToUserProfile(forceReinstallCredentialProvider);
 
+#if WIF
+        const feedUrl = tl.getInput("feedUrl");
+        const entraWifServiceConnectionName = tl.getInput("workloadIdentityServiceConnection");
+
+        if (feedUrl && entraWifServiceConnectionName) {
+            if (url.parse(feedUrl)) {
```
Comment (zhenghao104) [inline Tasks/NuGetAuthenticateV1/main.ts:28]: This doesn't actually do anything, "https://test.com,https://test.com" will only pass.

File: Tasks/NuGetAuthenticateV1/main.ts [hunk 2]
```diff
@@ -1,29 +1,53 @@
...
+                tl.debug(tl.loc("Info_AddingFederatedFeedAuth", entraWifServiceConnectionName, feedUrl));
+                await configureEntraCredProvider(ProtocolType.NuGet, feedUrl, entraWifServiceConnectionName);
+                console.log(tl.loc("Info_SuccessAddingFederatedFeedAuth", feedUrl));
+                federatedFeedAuthSuccessCount++;
+                return;
```
Comment (zhenghao104) [inline Tasks/NuGetAuthenticateV1/main.ts:33]: Nit: just for consistency and readability, return after else {}

File: Tasks/NuGetAuthenticateV1/taskJsonOverride.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{
+    "inputs": [
+        {
+            "name": "forceReinstallCredentialProvider",
```
Comment (zhenghao104) [inline Tasks/NuGetAuthenticateV1/taskJsonOverride.json:4]: I think the order here matters for the task input user interface. Put service connection and feed URL at the top to be consistent with the other tasks.

File: Tasks/NuGetAuthenticateV1/main.ts (hunk 1 above)
Comment (embetten) [reply Tasks/NuGetAuthenticateV1/main.ts]: Removed - letting invalid urls fall through to auth failure

File: Tasks/NuGetAuthenticateV1/taskJsonOverride.json (hunk 3 above)
Comment (embetten) [reply Tasks/NuGetAuthenticateV1/taskJsonOverride.json]: done

File: Tasks/NuGetAuthenticateV1/main.ts (hunk 2 above)
Comment (embetten) [reply Tasks/NuGetAuthenticateV1/main.ts]: done

**Review by zhenghao104 — APPROVED** [review]

**Pull request merged**